### PR TITLE
Improve MOTD documentation

### PR DESCRIPTION
**Pull request by cpaelzer**
## Proposed Commit Message

Out of [discussions on discourse](https://discourse.ubuntu.com/t/remove-pro-advertising-from-apt/34051/2) and [discussions on issues](https://github.com/canonical/ubuntu-pro-client/issues/2458) I've realized that this part of the documentation is still weak.
So no more complaints, let us improve it!

I've put myself into a position of "what if I know nothing and would want to understand how it works and disable it".
And I must say that I'm quite pleased with the outcome.
I'm more than happy to take feedback, but think that this position I outlined about would now find all they need.
It is almost easy now to disable things and I'm not sure we need a `pro ...` command to make it even easier.
Well, that is up for discussion anyway ...

I wasn't able to fully cover #2458 as this would require splitting the update-notifier from the pro/esm messages.
But all other things that came up so far should hopefully be much improved by this.

And if I wrote something wrong, well that is great as this PR is about improving documentation - let us discuss and iterate on it in that case.

```
Improve MOTD documentation

This covers:
- Add further outputs we print to MOTD messaging.
- Clarify terms potentially unknown to readers
- Structure headers and paragraphs for symetry
- add info on esm-announce
- Add info on how to disable the messages

This is related to, but not fully fixing #2458 yet.

Fixes: #2533
```

## Test Steps

This is a documentation-only change describing the status as-is.
Test = reviewing and rendering the new doc content.

## Checklist

 - [n/a] I have updated or added any unit tests accordingly
 - [n/a ] I have updated or added any integration tests accordingly
 - [x] I have updated or added any documentation accordingly

## Does this PR require extra reviews?

 - [ ] Yes
 - [x] No

tags:
NO-JIRA NO-LP

### CODE REVIEW

File: docs/explanations/motd_messages.md
Comment (s-makin) [inline docs/explanations/motd_messages.md:6]: Once we're done tweaking the doc, could we add a bullet list here with the three sources? (I know the doc isn't long, but it's a helpful thing to have)

File: docs/explanations/motd_messages.md
Comment (s-makin) [inline docs/explanations/motd_messages.md:7]:
```suggestion
## MOTD messages about available updates
```

Comment (s-makin) [reply docs/explanations/motd_messages.md:7]: "MOTD message" or "MOTD messages" is more likely to be a search term, I think

File: docs/explanations/motd_messages.md
Comment (s-makin) [inline docs/explanations/motd_messages.md:14]: Can it be both?

File: docs/explanations/motd_messages.md
Comment (s-makin) [inline docs/explanations/motd_messages.md:19]:
```suggestion
`update-notifier` has always been used to add information about potential
```

File: docs/explanations/motd_messages.md
Comment (s-makin) [inline docs/explanations/motd_messages.md:20]:
```suggestion
updates to MOTD to raise user awareness. With the advent of Ubuntu Pro they are
```

File: docs/explanations/motd_messages.md
Comment (s-makin) [inline docs/explanations/motd_messages.md:24]:
```suggestion
unreadable output as it is meant for program use. You can add `--human-readable`
```

File: docs/explanations/motd_messages.md
Comment (s-makin) [reply docs/explanations/motd_messages.md:25]: Implied that the information you see in MOTD is different when using the `--human-readable` flag

File: docs/explanations/motd_messages.md
Comment (s-makin) [inline docs/explanations/motd_messages.md:29]:
```suggestion
On a machine that runs an Ubuntu release for which the `esm-apps` service
```

File: docs/explanations/motd_messages.md
Comment (s-makin) [inline docs/explanations/motd_messages.md:30]:
```suggestion
is available, but not yet attached to an Ubuntu Pro subscription, there will
```

File: docs/explanations/motd_messages.md
Comment (s-makin) [inline docs/explanations/motd_messages.md:31]:
```suggestion
be a message notifying the user that there may be more security updates available through ESM.
```

Comment (s-makin) [reply docs/explanations/motd_messages.md:31]: Although "hint" isn't wrong, it may have some connotations that we don't intend to imply. I think message would generally be clearer and have less room for interpretation.

File: docs/explanations/motd_messages.md
Comment (s-makin) [inline docs/explanations/motd_messages.md:45]:
```suggestion
In the opposite situation, if an LTS machine has the `esm-infra` and `esm-apps` services enabled then users will see the following output in MOTD:
```

File: docs/explanations/motd_messages.md
Comment (s-makin) [inline docs/explanations/motd_messages.md:57]: I got a bit confused by the wording in this section. I wasn't sure why, if the machine is attached and in ESM, why only ESM infra being advertised. I asked about this earlier and it was confirmed that whichever services are enabled get shown but I'm not sure that's clear here. We probably want to revise this wording to make sure it reflects what's currently happening after all the changes to terminology.

(I should clarify, I attached the comment to the sub-heading and not the paragraph below it by mistake, sorry!)

File: docs/explanations/motd_messages.md
Comment (s-makin) [inline docs/explanations/motd_messages.md:143]:
```suggestion
When Ubuntu Pro became generally available, a temporary announcement was made
through MOTD. This was intended to raise awareness of Pro now being available
and free for personal use, and was shown on systems covered by `esm-apps`. 
```

File: docs/explanations/motd_messages.md
Comment (s-makin) [inline docs/explanations/motd_messages.md:153]: Since this message was intended as a limited-time announcement to coincide
with the release of Ubuntu Pro into general availability, it was removed in
27.14.

File: docs/explanations/motd_messages.md
Comment (s-makin) [inline docs/explanations/motd_messages.md:154]:
```suggestion
## How are these messages inserted into MOTD and how can I disable them?
```

Comment (s-makin) [reply docs/explanations/motd_messages.md:154]: "can" is more likely to be searched for

File: docs/explanations/motd_messages.md
Comment (s-makin) [inline docs/explanations/motd_messages.md:157]:
```suggestion
Just as there are different purposes to the messages outlined above,
there are different sources producing the MOTD that one sees at login.
```

File: docs/explanations/motd_messages.md
Comment (s-makin) [inline docs/explanations/motd_messages.md:163]:
```suggestion
to disable them, but you can selectively disable them via the config
files that add them, as outlined below.
```

File: docs/explanations/motd_messages.md
Comment (s-makin) [inline docs/explanations/motd_messages.md:165]: I'm not sure what "those" are being referred to here - please clarify

File: docs/explanations/motd_messages.md
Comment (s-makin) [inline docs/explanations/motd_messages.md:170]:
```suggestion
related packages. If you later realise that you want them back, you will need
to reinstall the related packages and tell `apt`/`dpkg` to offer you to restore
those files using the following command:

```
sudo apt install --reinstall -o Dpkg::Options::="--force-confask" ubuntu-advantage-tools update-notifier-common```
```

Comment (s-makin) [reply docs/explanations/motd_messages.md:170]: Well...that didn't work as successfully as I'd hoped :P

Comment (s-makin) [reply docs/explanations/motd_messages.md:170]: Not sure why it wouldn't let me add a code block there, but for the sake of tidiness I like to have code snippets on a separate line (especially if they're long snippets)

File: docs/explanations/motd_messages.md
Comment (s-makin) [inline docs/explanations/motd_messages.md:172]: If we change the header at the top, this should also be changed to match

**Review by s-makin — COMMENTED** [review]
Round 1 of comments/suggestions from me completed :) we will probably need to do a sweep to make sure that the language we're using throughout is still correct following the recent evolutions in terminology around ESM. My own knowledge of this is still a bit too shaky to be able to provide any suggestions around that.

File: docs/explanations/motd_messages.md
Comment (cpaelzer) [reply docs/explanations/motd_messages.md:19]: has always added

File: docs/explanations/motd_messages.md
Comment (cpaelzer) [reply docs/explanations/motd_messages.md:142]: systems that could be covered by

File: docs/explanations/motd_messages.md
Comment (cpaelzer) [reply docs/explanations/motd_messages.md:156]: the MOTD -> "these MOTD elements"

File: docs/explanations/motd_messages.md
Comment (cpaelzer) [reply docs/explanations/motd_messages.md:181]: => subscription

File: docs/explanations/motd_messages.md
Comment (cpaelzer) [reply docs/explanations/motd_messages.md:6]: ack and done

File: docs/explanations/motd_messages.md
Comment (cpaelzer) [reply docs/explanations/motd_messages.md:60]: ack

File: docs/explanations/motd_messages.md
Comment (cpaelzer) [reply docs/explanations/motd_messages.md:142]: done

File: docs/explanations/motd_messages.md
Comment (cpaelzer) [reply docs/explanations/motd_messages.md:152]: done

File: docs/explanations/motd_messages.md
Comment (s-makin) [reply docs/explanations/motd_messages.md:23]: For some reason it left this line in here, so it's rendering both this and the following one (the one we edited it to)

File: docs/explanations/motd_messages.md
Comment (s-makin) [inline docs/explanations/motd_messages.md:65]:
```suggestion
years of support") Ubuntu releases, where the hint is about ESM Apps
```

File: docs/explanations/motd_messages.md
Comment (s-makin) [inline docs/explanations/motd_messages.md:66]:
```suggestion
extending the coverage to the Universe repositories.
```

File: docs/explanations/motd_messages.md
Comment (s-makin) [inline docs/explanations/motd_messages.md:68]: This whole section is much clearer now, thank you!

File: docs/explanations/motd_messages.md
Comment (s-makin) [inline docs/explanations/motd_messages.md:110]: I'm not sure what "close to" means in this context - is there a more specific word that we can use instead?

**Review by s-makin — COMMENTED** [review]
Couple of small nits, but otherwise looks good. I'll hold off on an approving review for now until after Lucas has had a chance to review it.

**Review by lucasmoura — APPROVED** [review]
I could not find any issues with the new addition to the MOTD doc

**Review by s-makin — APPROVED** [review]

File: docs/explanations/motd_messages.md
Comment (cpaelzer) [reply docs/explanations/motd_messages.md:64]: done

File: docs/explanations/motd_messages.md
Comment (cpaelzer) [reply docs/explanations/motd_messages.md:65]: done

File: docs/explanations/motd_messages.md
Comment (cpaelzer) [reply docs/explanations/motd_messages.md:109]: yep, "... delivered next to the content created by `update-notifier`."
Done

**Comment by cpaelzer** [comment]
I have addressed the nit-picks of the second round. With @s-makin and @lucasmoura happy I guess we can land this now?

**Comment by s-makin** [comment]
> I have addressed the nit-picks of the second round. With @s-makin and @lucasmoura happy I guess we can land this now?

I think so!

**Pull request merged**
